packages/vite/src/node/plugins/worker.ts

```diff
@@ -49,10 +49,32 @@ function saveEmitWorkerAsset(
   workerMap.assets.set(fileName, asset)
 }
 
+// Ensure that only one rollup build is called at the same time to avoid
+// leaking state in plugins between worker builds.
+const workerConfigSemaphore = new WeakMap<
```

Comment:
Would be great to put a TODO here so we come back and try to parallelize it again. But otherwise neat trick for now 👍